spectator-api/src/main/java/com/netflix/spectator/api/patterns/LongTaskTimer.java

```diff
@@ -68,8 +68,14 @@ public static LongTaskTimer get(Registry registry, Id id) {
   private LongTaskTimer(Registry registry, Id id) {
     this.clock = registry.clock();
     this.id = id;
-    registry.monitorValue(id.withTag(Statistic.activeTasks), this, LongTaskTimer::activeTasks);
-    registry.monitorValue(id.withTag(Statistic.duration),    this, t -> t.duration() / NANOS_PER_SECOND);
+    PolledGauge.using(registry)
+        .withId(id)
+        .withTag(Statistic.activeTasks)
+        .monitor(this, LongTaskTimer::activeTasks);
```

Comment:
The fields have already been initialized when this is called so it shouldn't be a problem. In this case they can be moved after the object is constructed though as the construction is limited to a particular method.